File: src/main/java/leader/input/Input.java
package leader.input;

import java.io.Scanner;

public class Input
{	
	private Scanner _scanner;
	
	public Input()
	{
		_scanner = new Scanner( System.in );
	}
	
	public InputCommand getInput( String prompt )
	{
		System.out.print( prompt );
		String inp = _scanner.nextLine().trim();
		// attempt to parse it and construct command, if unable, return raw input
		InputCommand cmd = InputCommand.parse( inp );
		return cmd;
	}
	
	public void waitForInput()
	{
		System.out.print( "(Press Enter to Continue)" );
		_scanner.nextLine();
	}
}

File: python/input/input.py


class Input( object ):
	def __init__( self ):
		self.command = Command()
	
	
	def get_input( self, prompt="> " ):
		inp = raw_input( prompt ).strip()
		# attempt to parse it and construct command, if unable, return raw input
		cmd = self.parse( inp )
		return inp if cmd is None else cmd
	
	
	def parse( self, inp ):
		cmdobj = self.command.parse( inp )
        return cmdobj
	